qemu/tests/win_vioser_test.py

```diff
@@ -0,0 +1,78 @@
+import logging
+import re
+import select
+
+from virttest import error_context
+from virttest import utils_test
+from virttest import utils_misc
+from virttest.utils_virtio_port import VirtioPortTest
+from qemu.tests.virtio_driver_sign_check import get_driver_file_path
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    virtio serial test for windows vm:
+    1) Boot guest with virtioserialport;
+    2) Transefer data from guest to host;
+    3) Transefer data from host to guest.
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+
+    def transfer_from_host_to_guest():
+        port.open()
+        port.sock.setblocking(0)
+        port.sock.sendall(transfer_data)
+        output = session.cmd_output(guest_receive_cmd)
+        if not re.findall(guest_pattern, output, re.M):
+            test.fail("Guest fails to receive data, output is: %s" % output)
+        port.close()
+
+    def transfer_from_guest_to_host():
+        port.open()
+        port.sock.setblocking(0)
+        output = session.cmd_output(guest_send_cmd)
+        if params.get("check_from_guest", "no") == "yes":
+            if not re.findall(guest_pattern, output, re.M):
+                test.fail("Guest fails to send data, output is %s" % output)
+        elif select.select([port.sock], [], [], 1.0):
```

Comment:
This line seems meaningless, if you do not handle the unexpected result(port not ready or not readable), if we assume it is unlikely to happen then this line is more meaningless.

Reply:
Yes, you are right.
Modified, thanks